action.yml

```diff
@@ -54,6 +54,15 @@ runs:
         fi
         echo "MAVEN_OPTS=$MAVEN_OPTS" >> $GITHUB_ENV
 
+    - name: Prepare maven arguments
+      shell: bash
+      run: |
+        if [[ ${{ inputs.build-modules }} != '' ]]; then
+          echo "ADDITIONAL_MAVEN_ARGUMENTS=--also-make" >> $GITHUB_ENV
```

Comment:
Maybe better would be create new input e.g.: "build-dependent-modules" or some more meaningfull  name with default: true which will use "--also-make-dependents" otherwise: -also-make.

@mmichna Any thoughts? I know that you wanted to use this parameter for budiling only changed modules.

Reply:
i used it, but it have limitation because identification of changed file was problematic. But now we can use existing external GH action which can provide name of change module.

Problem is if maven repository GH cache file is removed and you will rerun PR with changed some sub-module. Then your build will fails because cache is missing. Maybe we can use: https://github.com/skjolber/maven-cache-github-action/blob/master/action.yml#L15

and then we can identify case when cache is empty. So they can rebuild all modules. But we need check it if there aren't another problems

@rbrisuda, @MartinMitro: ??? then we can again enable functionality for PR or also build workflows where we will rebuild only changed modules

Reply:
@rbrisuda: PLS check original @LubomirS question. I agree

Reply:
@rbrisuda: ???